UnicornEngine/video/include/unicorn/video/Graphics.hpp

```diff
@@ -0,0 +1,187 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_GRAPHICS_HPP
+#define UNICORN_VIDEO_GRAPHICS_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+#include <unicorn/system/WindowHint.hpp>
+
+#include <string>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace unicorn
+{
+namespace system
+{
+class Manager;
+class Monitor;
+class Window;
+}
+
+namespace video
+{
+class Renderer;
+
+enum class DriverType
+{
+    Vulkan
+};
+
+/** @brief Abstract graphics renderer
+ *
+ *  Initializes graphics system and proxies render calls to the renderer
+ */
+class Graphics
+{
+public:
+    /** @brief Constructs an empty graphics system */
+    Graphics(system::Manager& manager);
+
+    /** @brief  Destructs a graphics system
+    *
+    *  Also calls Deinit() to release all resources
+    *
+    *  @sa Deinit()
+    */
+    ~Graphics();
+
+    Graphics(const Graphics& other) = delete;
+    Graphics(const Graphics&& other) = delete;
```

Comment:
I'm not sure if it's relevant in this particular case. But move constructor does not have to take const rvreference since it's going to mutate `other` object by moving staff from it.